Automattic/wp-calypso
# Themes Showcase: Replace empty search card with more theme options

Fixes https://github.com/Automattic/dotcom-forge/issues/1453

#### Proposed Changes

Replaces the previous upsell nudge displayed on empty searches, with a new card that offer users to perform the following actions:
- Design your own theme / homepage: Allow users to change their own homepage in the site editor (only in the logged-in showcase, and only if the site has a block theme active).
- Do it for me: Upsells the DIFM service.
- Upload a theme: Upsells the Business plan and/or nudges users to upload a third-party/custom theme missing.

**Screenshots**
<table>
<thead><tr><th>Conditions</th><th>Before</th><th>After</th></tr></thead>
<tbody>
<tr>
<td>Showcase: Logged-in<br><br>Plan: Free<br><br>Current theme: Block theme</td>
<td>
<img width="1001" alt="Screenshot 2023-02-02 at 14 49 25" src="https://user-images.githubusercontent.com/1233880/216342469-a324fc56-c6b2-4389-b608-a2987d88fa0f.png">
</td>
<td>
<img width="1006" alt="Screenshot 2023-02-03 at 11 15 23" src="https://user-images.githubusercontent.com/1233880/216574321-4d11c7b9-ce6c-4b95-980e-07f9e36aa4e5.png">
</td>
</tr>
<tr>
<td>Showcase: Logged-in<br><br>Plan: Business<br><br>Current theme: Block theme</td>
<td>
<img width="1005" alt="Screenshot 2023-02-02 at 14 52 14" src="https://user-images.githubusercontent.com/1233880/216343142-7ce3f302-0b86-4371-be7a-e30db5e51b93.png">
</td>
<td>
<img width="1006" alt="Screenshot 2023-02-03 at 11 16 00" src="https://user-images.githubusercontent.com/1233880/216574487-b578f337-65e1-467c-be0f-d1c2d659dcfe.png">
</td>
</tr>
<tr>
<td>Showcase: Logged-in<br><br>Plan: Business<br><br>Current theme: Legacy theme</td>
<td>
<img width="1005" alt="Screenshot 2023-02-02 at 14 52 14" src="https://user-images.githubusercontent.com/1233880/216343142-7ce3f302-0b86-4371-be7a-e30db5e51b93.png">
</td>
<td>
<img width="1005" alt="Screenshot 2023-02-03 at 11 28 35" src="https://user-images.githubusercontent.com/1233880/216578168-3d538ba2-b105-4101-803b-fb8ade0bdbc8.png">
</td>
</tr>
<tr>
<td>Showcase: Logged-out</td>
<td>
<img width="1279" alt="Screenshot 2023-02-02 at 14 56 13" src="https://user-images.githubusercontent.com/1233880/216343973-49316e06-3f2b-48ab-96cf-241ae9a464b8.png">
</td>
<td>
<img width="1278" alt="Screenshot 2023-02-02 at 14 56 59" src="https://user-images.githubusercontent.com/1233880/216344166-226838ac-37a1-4bdc-bac7-44cbe9cf1a74.png">
</td>
</tr>
</tbody>
</table>

**Events**

The following events a tracked when a user clicks on any of the options:
- Design your own theme / homepage: 
  - Event: `calypso_themeshowcase_more_options_design_homepage_click`
  - Props:
    - `site_plan`: The site plan.
    - `search_term`: The term searched.
    - `destination`: Where we are sending the user to. The only possible value right now is 'site-editor', but it can be extended in the future once we can start the Pattern Assembler.
- Do it for me:
  - Event: `calypso_themeshowcase_more_options_difm_click`
  - Props:
    - `site_plan`: The site plan.
    - `search_term`: The term searched.
- Upload a theme:
  - Event: `calypso_themeshowcase_more_options_upload_theme_click`
  - Props:
    - `site_plan`: The site plan.
    - `search_term`: The term searched.
    - `destination`: Where we are sending the user to. Possible values are `signup` (for the logged-out showcase), `upload-theme` if the site has an eligible plan, or `checkout` if the site doesn't have an eligible plan.

#### Testing Instructions

- Use the Calypso live link below
- Switch to a Free site with a block theme active
- Go to Appearance > Themes
- Enter a term that doesn't produce any result
- Observe the new card
- Make sure it includes 3 options: Design your own homepage, Do it for me, Upload theme
- Make sure it looks good and the copy is clear
- Make sure all actions redirect to the appropriate place
- Make sure it is responsive on mobile viewports
- Switch to a Simple site with a Business plan and a block theme active
- Make sure the "Upload theme" action no longer upsells the Business plan
- Make sure the "Upload theme" action sends users to `/theme/upload/:site`
- Activate a legacy theme
- Make sure the "Design your own homepage" action is no longer visible
- Make sure the "Upload theme" action sends users to `/wp-admin/theme-install.php`
- Switch to an Atomic site with a Business plan
- Open the `/themes` path on an incognito window
- Enter a term that doesn't produce any result
- Make sure it includes 2 options: Do it for me, Upload theme
- Make sure the "Upload theme" action initiates the Business plan signup flow
